scripts/setup-host.sh

```diff
@@ -27,6 +27,26 @@ check_environment() {
         msg "NOTE: external_address has not been set in ${git_root}/configuration/config.toml. You might need it if your external IP address is different."
         msg
     fi
+
+    local shared_lib_path
+    local shared_lib_env_varible
+    shared_lib_env_varible="LD_LIBRARY_PATH"
+    shared_lib_path=${LD_LIBRARY_PATH:-}
+
+    if [ "${os}" == "darwin" ]; then
+        msg "NOTE: WasmVM don't have a shared library on MacOS"
+        msg "NOTE: Also, due to System Integrity Protection from MacOS, dynamic linking variable DYLD_LIBRARY_PATH are purged when launching protected processes"
+        msg "NOTE: please compile axelard binary by staticially linking libwasmvm to run axelar node on MacOS"
+        die "Script don't support MacOS"
+    fi
+
+    if [ -z "${shared_lib_path}" ]; then
+        die "${shared_lib_env_varible} is not set. Run export ${shared_lib_env_varible}=\"${share_lib_directory}\""
+    fi
+
+    if [[ $shared_lib_path != *"${share_lib_directory}"* ]]; then
+        die "NOTE: ${share_lib_directory} missing from ${shared_lib_env_varible}. Run export ${shared_lib_env_varible}=\"\${${shared_lib_env_varible}}:${share_lib_directory}\""
```

Comment:
This should be set to the wasm download folder and exported for the user. Otherwise they need to make infra changes for running the scripts

Reply:
yes, the message displayed is set to the directory where wasm is downloaded.  It will be something like 

```
FAILED: NOTE: /root/.axelar_testnet/lib missing from LD_LIBRARY_PATH. Run export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/root/.axelar_testnet/lib"
```
I think its better to display message and let users decide how they want to persist LD_LIBRARY_PATH, just like what we're doing with `ulimit -n`.

Reply:
`ulimit` has global implications. But modifying ld path is part of running the node. Without this, we'd also require everyone to update their infra before upgrade, which causes a major headache with the rollout. Let's set it by default, we can change it if validators prefer.

Reply:
ulimit cmd is scoped to the current shell, it doesn't persist for the user globally or for future shells unless its set in `/etc/security/limits.conf` or `~/.bashrc `. Same goes with export cmd and environment variables. 

In general, its not a good idea setting these kinda configs from a shell script, because it will require sudo access if we set it globally in /etc/environment or if we do set it in `~/.bashrc` and source it from the script, the change will not bubble up and persist in the parents shell anyways.  Its a hassle but I think asking user to do this manually from the getgo is the fool proof way.

Reply:
ah, yes, forgot it's limited to the shell. I understand that it's not ideal to use a default here. But it's also not ideal to use these setup scripts vs the manual binary setup we have recommended to validators. I'd really prefer to avoid validators/node operators running into consensus halt due to this config. We can ask validators to set it up themselves and override the default in the docs, but it would be good to have this as a fallback. It's not really needed for the parent shell anyways, just for running the node itself.